new-log-viewer/src/services/LogExportManager.ts

```diff
@@ -0,0 +1,70 @@
+import {downloadBlob} from "../utils/file";
+
+
+const EXPORT_LOG_PROGRESS_INITIALIZATION = 0;
+const EXPORT_LOG_PROGRESS_COMPLETE = 1;
+
+/**
+ * Manager for exporting logs to a file.
+ */
+class LogExportManager {
+    /**
+     * Internal buffer which stores decoded chunks of log data.
+     */
+    readonly #chunks: string[] = [];
+
+    /**
+     * Total number of chunks to export.
+     */
+    readonly #numChunks: number;
+
+    /**
+     * Name of the file to export to.
+     */
+    readonly #exportedFileName: string;
+
+    constructor (numChunks: number, fileName: string) {
+        this.#numChunks = numChunks;
+        this.#exportedFileName = `exported-${fileName}-${new Date().toISOString()
+            .replace(/[:.]/g, "-")}.log`;
```

Comment:
* Can we change to putting the timestamp after `exported-` rather than nesting the filename between generated strings? That makes it easier for the user to strip off the generated string if they want.
* Can we get rid of the `.log` suffix so that the file remains open-able by the log viewer?

Reply:
> * Can we change to putting the timestamp after exported- rather than nesting the filename between generated strings? That makes it easier for the user to strip off the generated string if they want.

Sounds better.

> Can we get rid of the .log suffix so that the file remains open-able by the log viewer?

Not sure if I get this. As Henry pointed out, without the filename extension correctly set, file editors wouldn't easily know how to interpret the format. On the other hand, having the original file extension would cause confusion to the log viewer as well, since we currently don't check the magic numbers to determine the type yet. Are you suggesting that we want to use a `.txt` extension instead?

Reply:
* I think putting the timestamp after `exported-` doesn't affect Junhao's request?
* The `.log` suffix can't be opened by the log viewer which I think is more important than whether a Windows user can open the file by double-clicking it, lol. Also the files I'm testing with have a `.jsonl` extension, so it's weird that I can open a `.jsonl` file in the log viewer, but I can't open the exported version of it.

Reply:
(the comments are posted out of orders lol.) For the later point regarding file name extensions, let me further clarify a bit.
Say that we're exporting logs from a IR stream archive with original file name extension `.clp.zst`. If we don't append a `.txt` or `.log` extension to the exported file name, it would appear that the exported file is still in the CLP IR stream format and cause confusions, right?

Reply:
Ah, right, sorry. I think we need more intelligent extension creation then.

* If exporting a file as the same type, leave the extension or add one if it doesn't already have one.
* If exporting a file as a different type, change the file's extension (or if it doesn't have a known extension, add the correct extension).

Reply:
Clarified in an offline discussion that the exported file is always a `.log` (even a `.jsonl` is reformatted as a `.log` by formatting the messages). I was mistakenly confusing exporting compressed and uncompressed files.

So for now, we decided to leave the `.log` extension.